docs/developer-docs/integrations/sns/get-sns/local-testing.md

```diff
@@ -134,54 +241,54 @@ If the proposal is rejected, the dapp canisters' controllers are automatically s
 back to the developer principals that you
 have defined in the [initialization file](preparation.md).
 
-#### 10. Test sale participation
-After the sale has been started by the NNS, test that you can participate in the sale
-as expected.
-To do so, use your `sns-quill` principal `identityDevNeuron`
-and learn what command to use `here` <!-- TODO: SNS quill must allow sale participation.-->
+#### 9. Test sale participation
+After the sale has been started by the NNS, test that you can participate in the sale as expected.
+Return to the launchpad section in the NNS dapp and hit refresh.
+You should now see the SNS move into the "Current Launches" section.
+If you click on it, you will be able to read details about the project.
 
-To finish the sale and proceed with other testing, you can either wait for the sale 
-deadline to exceed or you can participate in the sale repeatedly until you hit the
-maximum ICP that the sale accepts.
+Note the sale start time. Wait until then, then hit refresh. You should now see an interface to buy SNS tokens.
+Buy some tokens. The sale will be complete when either the maximum investment has been reached or the sale end time is reached.
+If you use the default SNS configuration you can buy all 50 ICP.
+This is convenient for testing but in a real SNS you may wish to limit the stake so that no investor has excessive influence over the project.
 
-#### 11. Repeatedly test upgrading the dapp canister(s) and adding more canisters.
+You can then check the neurons of the SNS:
+
+``` bash
+dfx canister call sns_governance list_neurons '(record {limit= 20})'
+```
+
+#### 10. Repeatedly test upgrading the dapp canister(s) and adding more canisters.
 You might want to repeatedly test that you can upgrade the dapp at all stages.
-As already explained in [Step7](#7-test-upgrading-the-dapp-canisters-by-sns-proposal),
+As already explained in [Step 6](#6-test-upgrading-the-dapp-canisters-by-sns-proposal),
 you might want to test this before the decentralization sale and you might also want to test this
 during and after the decentralization sale.
 To do so, you can repeat the instructions from
-[Step 7](#7-test-upgrading-the-dapp-canisters-by-sns-proposal).
+[Step 6](#6-test-upgrading-the-dapp-canisters-by-sns-proposal).
 
 You can also test adding new dapp canisters under the SNS control at different stages,
-by additional SNS proposals as explained in [Step 6](#6-register-the-dapp-in-the-sns).
+by additional SNS proposals as explained in [here](https://github.com/dfinity/sns-quill#submit-a-proposal).
+For example, you could make a proposal to set the minimum neuron stake like this:
```

Comment:
I think this example does not fit the explanation before that.
The explanation states that you can register different dapps (as explained in step 5) and the example is about changing parameters.
I would probably just link back to Step 5?
The reason is that during the decentralization sale, the governance cansiter is in a special "pre-genesis" mode and the proposal to change the minimum neuron stake should not work. This is why I would move this concrete example to later.

Reply:
Actually, I would put this example to 11, where it would be perfect.